src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
+
+    this.logger.debug(
+      `Request Method:[${req.method}] || Request URI:${
+        req.baseUrl
+      } || Request Body:[${JSON.stringify(
+        req.body,
+      )}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${
+        req.ip
+      }] || User Details: [${JSON.stringify(userDetail)}]`,
+    );
+    next();
```

Comment:
PullProofAI

The `LoggerMiddleware` class implementation looks good. It captures request details and logs them appropriately. However, there are a few suggestions to enhance the middleware:
1. **Error Handling**: Add error handling to catch and log any errors that might occur during the logging process.
2. **Type Checking**: Ensure proper type checking for `req.headers`, `req.body`, `req.query`, and other properties to avoid runtime errors.
3. **Security**: Be cautious with logging sensitive information like user details. Consider masking or omitting certain sensitive data for security reasons.

```diff
+ // Add error handling
+ try {
+     // Existing code block
+ } catch (error) {
+     this.logger.error(`Error occurred in logging: ${error}`);
+ }
+
+ // Type checking for req.headers, req.body, req.query, etc.
+ const token = (req.headers?.authorization || '').split(' ')[1] || '';
+ const userDetail = this.jwt.decode(token) || {};
+
+ // Security consideration for logging sensitive information
+ // Mask or omit sensitive data before logging if needed
```


<!-- This is an auto-generated comment by PullProofAI -->